superset-frontend/src/views/CRUD/data/dataset/AddDataset/index.tsx

```diff
@@ -91,22 +90,22 @@ export default function AddDataset() {
       })
     : undefined;
 
-  const getDatasetsList = async () => {
-    await UseGetDatasetsList(queryParams)
-      .then(json => {
-        setDatasets(json?.result);
-      })
-      .catch(error =>
-        logging.error('There was an error fetching dataset', error),
-      );
-  };
+  const { getDatasetsList, datasets, datasetNames } =
+    useGetDatasetsList(queryParams);
 
   useEffect(() => {
     if (dataset?.schema) {
       getDatasetsList();
     }
   }, [dataset?.schema]);
 
+  const id = window.location.pathname.split('/')[2];
+  useEffect(() => {
+    if (!Number.isNaN(parseFloat(id))) {
```

Comment:
```suggestion
    if (!Number.isNaN(parseInt(id, 10))) {
```

Reply:
Fixed in [`this commit`](https://github.com/apache/superset/pull/22043/commits/41fbaeb11bfd574a84a24732cadd2dae25e6f39d).